03-spatial-joins.ipynb

```diff
@@ -239,6 +239,92 @@
     "joined['continent'].value_counts()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "**Exercise**: Which rivers pass within 1 degree of Paris? And for Hong Kong?\n",
+    "\n",
+    "Tip: `rivers` contains empty geometries, make sure to get rid of them before anything else\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!--\n",
+    "paris = cities.query('name == \"Paris\"')\\\n",
+    "              ['geometry']\\\n",
+    "              .squeeze()\\\n",
+    "              .buffer(1)\n",
+    "\n",
+    "river_lines = rivers[rivers.geometry.notna()]\n",
+    "\n",
+    "river_lines[river_lines.crosses(paris)]\n",
+    "\n",
+    "hk = cities.query('name == \"Hong Kong\"')\\\n",
+    "              ['geometry']\\\n",
+    "              .squeeze()\\\n",
+    "              .buffer(1)\n",
+    "\n",
+    "river_lines = rivers[rivers.geometry.notna()]\n",
+    "\n",
+    "river_lines[river_lines.crosses(hk)]\n",
+    "-->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "**Exercise** [PRO]: Find all European cities within 1 degre of at least one river\n",
+    "\n",
+    "\n",
+    "This requires:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Obtaining a list of European cities\n",
+    "1. Creating a `GeoDataFrame` with all river lines\n",
+    "1. For every European city, checking whether at least one river \"crosses\" the area within 1 degree\n",
```

Comment:
Isn't this an exercise for notebook 2 ?

Reply:
A list of European cities requires a spatial join. It's a compound exercise.